### PR TITLE
issue_6 Create/Edit Admin hotel

### PR DESCRIPTION
**Pull request by loveisalie92**
#6
Create controller file for admin hotel
Create FormRequest for create and update admin hotel
Create routes
Create views
Create tests for Admin hotel include create/edit

### CODE REVIEW

**Comment by maivien310** [comment]
- Pull request title is not understandable, title should show the main content of the PR
- Description : no need to list file changes, you should show what you did in this PR

**Comment by maivien310** [comment]
Please check coding style for your pull request 
[http://www.php-fig.org/psr/psr-2/](http://www.php-fig.org/psr/psr-2/)

File: app/AdminHotel.php [hunk 1]
```diff
@@ -7,10 +7,19 @@
 class AdminHotel extends Model
 {
     //
-    protected $table='admin_hotels';
+    protected $table = 'admin_hotels';
 
-    protected $fillable = ['hotel_id', 'username', 'password', 'name', 'email', 'phone'];
-
-    protected $hidden = ['password', 'remember_token'];
+    protected $fillable = [
```
Comment (maivien310) [inline app/AdminHotel.php]: Check style here

File: app/Http/Controllers/Admin/AdminHotelController.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        
+    }
+
+    /**
+     * Show the form for creating a new resource.
```
Comment (maivien310) [inline app/Http/Controllers/Admin/AdminHotelController.php]: "Show the form for creating a new resource." This description is generated by Laravel, pls change it

File: app/Http/Controllers/Admin/AdminHotelController.php [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+     * @param  \Illuminate\Http\Request\Admin\
+     *          AdminHotelCreateFormRequest  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(AdminHotelCreateFormRequest $request)
+    {
+
+        $data = $request->all();
+        if (AdminHotel::create($data)) {
+            Session::flash('flash_message', trans('messages.create_success'));
```
Comment (maivien310) [inline app/Http/Controllers/Admin/AdminHotelController.php]: The key is `create_success` but the content is `You had successfully created an account`. What happen if you want to add new message when create another entity like `hotel`, `room_types` ....

File: app/Http/Controllers/Admin/AdminHotelController.php [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+     * @param  \Illuminate\Http\Request\Admin\
+     *         AdminHotelUpdateFormRequest  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(AdminHotelUpdateFormRequest $request, $id)
+    {
+        $adminHotel = AdminHotel::findOrFail($id);
+        $adminHotel->update($request->all());
+        Session::flash('flash_message', trans('messages.edit_success'));
```
Comment (maivien310) [inline app/Http/Controllers/Admin/AdminHotelController.php]: `edit_success` for what ???

File: app/Http/Requests/Admin/AdminHotelCreateFormRequest.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace HotelBooking\Http\Requests\Admin;
+
+use HotelBooking\Http\Requests\Request;
+
+class AdminHotelCreateFormRequest extends Request
```
Comment (maivien310) [inline app/Http/Requests/Admin/AdminHotelCreateFormRequest.php]: Class should have the description

File: resources/lang/en/messages.php [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+return [
```
Comment (maivien310) [inline resources/lang/en/messages.php]: check the coding style

File: database/seeds/DatabaseSeeder.php [hunk 7]
```diff
@@ -14,7 +14,7 @@ public function run()
     {
         Model::unguard();
 
-        // $this->call(UserTableSeeder::class);
+        $this->call(MySeeder::class);
```
Comment (maivien310) [inline database/seeds/DatabaseSeeder.php]: class name `MySeeder`, it's very bad class name

File: resources/lang/jp/messages.php [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?php
+return [
```
Comment (maivien310) [inline resources/lang/jp/messages.php]: only English for now, remove `jp` folder

File: resources/views/admin/admin-hotel/create.blade.php [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            </div><!-- /.box-header -->
+                <!-- form start -->
+            <div class="box-body">
+                <div class="col-md-8 col-md-offset-2">
+                    {!! Former::vertical_open()
+                        ->method('POST')
+                        ->action(route('admin-hotel.store'))
+                    !!}
+                    @include('admin.messages.success')
+                    {!! Former::select('hotel_id')
```
Comment (maivien310) [inline resources/views/admin/admin-hotel/create.blade.php]: check style

File: resources/views/admin/messages/success.blade.php [hunk 10]
```diff
@@ -0,0 +1,7 @@
+@if (Session::has('flash_message'))
```
Comment (maivien310) [inline resources/views/admin/messages/success.blade.php]: the path to this file should be `resources/views/layout/admin/partials/flash.blade.php`

File: resources/views/admin_partials/navbar.blade.php [hunk 11]
```diff
@@ -1,7 +1,7 @@
 <nav class="navbar navbar-static-top" role="navigation">
```
Comment (maivien310) [inline resources/views/admin_partials/navbar.blade.php]: path to this file should be `resource/views/layouts/admin/partials/navbar.blade.php`

File: tests/ExampleTest.php [hunk 12]
```diff
@@ -0,0 +1,19 @@
+<?php
```
Comment (maivien310) [inline tests/ExampleTest.php]: Remove this file

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 13]
```diff
@@ -0,0 +1,317 @@
+<?php
+
+use Illuminate\Foundation\Testing\WithoutMiddleware;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use HotelBooking\AdminHotel;
+
+class AdminHotelTest extends TestCase
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: you're testing for `AdminHotelController` but the class name is `AdminHotelTest`

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+{
+
+	use DatabaseTransactions;
+	
+    /**
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: these lines are for what ?

File: tests/admin-hotel-tests/AdminHotelTest.php (hunk 13 above)
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: there are some tabs on this file

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 15]
```diff
@@ -0,0 +1,317 @@
...
+  	    $this->WithoutMiddleware();
+  	    $this->call('POST', route('admin-hotel.store'));
+  	    $this->assertResponseStatus(302);
+  	}
+
+  	/**
+     *Test if can get correct  hotel admin edit page status. 
+     * @return void
+     */
+  	public function testEditStatus()
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: truncate the table before testing

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 16]
```diff
@@ -0,0 +1,317 @@
...
+    public function testNewAdminHotelIsCreated()
+    {
+    	$this->visit(route('admin-hotel.create'))
+    		 ->type('hoteladmin892', '#username')
+    		 ->type('123456', '#password')
+    		 ->type('Justin Beiber', '#name')
+    		 ->type('testAdminHotel@gmail.com', '#email')
+    		 ->type('0987666223', '#phone')
+    		 ->press('Create')
+             ->seeIndatabase('admin_hotels', ['email' => 'testAdminHotel@gmail.com'])
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: why only `email` is tested, how about another columns

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 17]
```diff
@@ -0,0 +1,317 @@
...
+  	public function testEditStatus()
+  	{
+        $faker = Faker\Factory::create();
+        AdminHotel::create([
+            'hotel_id'=>rand(0,10),
+            'username'=>$faker->name,
+            'name' => $faker->name,
+            'email' => $faker->email,
+            'phone' => $faker->phoneNumber,
+            'password' => bcrypt(str_random(10)),
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: by this way `password` is hashed twice.

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 18]
```diff
@@ -0,0 +1,317 @@
...
+    public function testCreateAdminHotelWithoutUsername()
+    {
+    	$this->visit(route('admin-hotel.create'))
+    		 ->type('', '#username')
+    		 ->type('123456', '#password')
+    		 ->type('Justin Beiber', '#name')
+    		 ->type('testAdminHotel@gmail.com', '#email')
+    		 ->type('0987666223', '#phone')
+    		 ->press('Create')
+    		 ->see('The username field is required.');
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: language message ?

File: tests/admin-hotel-tests/AdminHotelTest.php [hunk 19]
```diff
@@ -0,0 +1,317 @@
...
+    }
+    /**
+     * Test if a hotel admin is edited
+     * @return [type] [description]
+     */
+    public function testAdminHotelIsEdited()
+    {
+        //factory(HotelBooking\AdminHotel::class)->make();
+        $faker = Faker\Factory::create();
+        AdminHotel::create([
```
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: Check the logic here

File: tests/admin-hotel-tests/AdminHotelTest.php (hunk 13 above)
Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: username unique hasn't tested

File: tests/admin-hotel-tests/AdminHotelTest.php (hunk 14 above)
Comment (loveisalie92) [inline tests/admin-hotel-tests/AdminHotelTest.php]: You mean 'use DatabaseTransactions' ?
This line for the database automating rollback after insert a record for a test

Comment (maivien310) [inline tests/admin-hotel-tests/AdminHotelTest.php]: No I mean the comment below

**Pull request merged**